feat(question): support linking questions

sosyz opened the pull request:
### Description
This PR introduces the ability to link questions, similar to how GitHub issues can be linked. This feature allows users to create connections between questions, enhancing the navigation and discoverability of related content.

### Changes
- Added a new API endpoint `/answer/api/v1/question/link` to fetch linked questions.
- Implemented the `QuestionLink` entity to store links between questions and answers.
- Updated the `QuestionController` and `QuestionService` to handle linking logic.
- Modified the frontend to display linked questions in the question detail view.
- Added tests for the new linking functionality.

### Related Issue
Closes [apache/incubator-answer#817](https://github.com/apache/incubator-answer/issues/817)

### Screenshots
![Linked Questions UI](https://github.com/user-attachments/assets/99fea926-e77b-48c9-98d6-b30fca8b80e7)
![Linked Questions UI](https://github.com/user-attachments/assets/b4fe6ea3-f04b-4dc1-b0f8-eae920c4cd9f)
![Linked Questions UI](https://github.com/user-attachments/assets/612e6662-8733-4f68-b4cb-1347d9c041f5)

### Additional Information
- Design file: [Linked sections](https://www.figma.com/design/MeOkbCEQS9jCDRVHcjTRtY/Answer-Core?node-id=88-472&t=ThKWBdYEX2pxbWm3-4)
- All linked questions page: [Figma Design](https://www.figma.com/design/MeOkbCEQS9jCDRVHcjTRtY/Answer-Core?node-id=24373-7193&t=ThKWBdYEX2pxbWm3-4)

sosyz commented:
Thank you for your feedback, @LinkinStars. I've made the following changes to address the issues you pointed out:

1. **Added title display for linked questions**
I've implemented the feature to show the title of linked questions on the linking page, as per the design.
<img width="938" alt="Linked page example" src="https://github.com/user-attachments/assets/04f6fc4a-7d64-44ea-aad0-25cf157bd163">

2. **Prevented linking a question to itself**
I've added a check to skip link addition in the database when the source and target question IDs are the same. However, I still save the ID link on the page. Perhaps a consistent behavior in page would be better?
These updates should resolve the main concerns you raised.

Please review these changes and let me know if any further adjustments are needed. I'm happy to make additional modifications to ensure this feature meets the project's requirements and design specifications.

shuashuai commented on ui/src/pages/Questions/Linked/index.tsx:
```diff
@@ -0,0 +1,127 @@
+  return (
+    <Row className="pt-4 mb-5">
+      <Col className="page-main flex-auto">
+        <h3>Linked Questions</h3>
```
add class .mb-3
<h3 className='mb-3'>Linked Questions</h3>

shuashuai commented on ui/src/pages/Questions/Linked/index.tsx:
```diff
@@ -0,0 +1,127 @@
+    <Row className="pt-4 mb-5">
+      <Col className="page-main flex-auto">
+        <h3>Linked Questions</h3>
+        <p>
```
add class .mb-5, and use <div>  replace <p> tag

 <div className='mb-5'>
          Questions linked to/from&nbsp;
          <a href={`/questions/${qid}`}>{questionTitle}</a>
        </div>

shuashuai commented on ui/src/pages/Questions/Detail/components/LinkedQuestions/index.tsx:
```diff
@@ -0,0 +1,126 @@
+  if (isLoading) {
+    return null;
+  }
+
```
When there is no data, the component will not be displayed directly.

shuashuai commented on ui/src/pages/Questions/Detail/components/LinkedQuestions/index.tsx:
```diff
@@ -0,0 +1,126 @@
+            </ListGroup.Item>
+          );
+        })}
+        {data.list.length === 0 ? (
```
Need to delete here

sosyz commented:
> No problems have been found so far, it's perfect, thanks for your contribution.

@robinv8 Thank you very much for your feedback!

sosyz commented:
@shuashuai @LinkinStars 
Hi~ I've addressed the feedback from code reviews and pushed the updates. Could you please take a look at the changes when you have a moment? Thanks!
